### PR TITLE
fix(gengapic): fix linkParser regexp to support multi-line link tags

### PR DESCRIPTION
**Pull request by quartzmo**
closes: #1097

### CODE REVIEW

**Comment by quartzmo** [comment]
@noahdietz Build error in compute-regen is:

```
[/home/runner/work/gapic-generator-go/gapic-generator-go] >>>> gofmt -s -d -w -l . <<<<
go: downloading github.com/googleapis/gax-go/v2 v2.5.0
go: downloading github.com/googleapis/gax-go v1.0.3
go: downloading golang.org/x/oauth2 v0.0.0-20220722155238-128564f6959c
go: downloading golang.org/x/net v0.0.0-20220802222814-0bcc04d9c69b
go: downloading google.golang.org/appengine v1.6.7
go: downloading golang.org/x/sys v0.0.0-20220804182731-e052cef7d300
go: downloading github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da
go: loading module retractions for github.com/googleapis/gax-go/v2@v2.0.1: parsing go.mod: go.mod:8:2: require google.golang.org/genproto: version "b98a9ff5e252" invalid: must be of the form v1.2.3
Error: Process completed with exit code 1.
```

Any suggestions?

**Comment by noahdietz** [comment]
@quartzmo uh oh...I think that means a bad pseudo-version was introduced into a go.mod/sum, but I'm not sure...I will look at it.

File: internal/gengapic/markdown_test.go [hunk 1]
```diff
@@ -73,11 +73,20 @@ func TestMDPlain(t *testing.T) {
...
+			in:   "html <a \n  href=\"/link/to/some#thing\">\n with a linebreak in tag</a> <br> test",
+			want: "html with a linebreak in tag (at /link/to/some#thing) \n test",
+		},
 		{
 			in:   "link to [a search engine](https://www.google.com) with request type [Search][foo.bar_baz.v1.Search], [biz][][buz][baz].",
 			want: "link to a search engine (at https://www.google.com) with request type Search, bizbuz.",
 		},
 	} {
+		defer func() {
+			if r := recover(); r != nil {
```
Comment (noahdietz) [inline internal/gengapic/markdown_test.go:86]: Why is this necessary?

Comment (quartzmo) [reply internal/gengapic/markdown_test.go:86]: Failure to match the regex results in a panic, adding this `Errorf` output allows you to see which test caused it.

Comment (quartzmo) [reply internal/gengapic/markdown_test.go:86]: It's a bit of a tradeoff since with this statement you don't get the whole backtrace, just the message for the panic.

#### Before

```
--- FAIL: TestMDPlain (0.00s)
panic: runtime error: index out of range [-1] [recovered]
	panic: runtime error: index out of range [-1]

goroutine 130 [running]:
testing.tRunner.func1.2({0x163c560, 0xc000042c00})
	/usr/local/go/src/testing/testing.go:1389 +0x24e
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1392 +0x39f
panic({0x163c560, 0xc000042c00})
	/usr/local/go/src/runtime/panic.go:838 +0x207
github.com/googleapis/gapic-generator-go/internal/gengapic.(*mdRenderer).html(0xc0004cc000, 0xc0006a16b0)
	/Users/chrisdsmith/oss/gapic-generator-go/internal/gengapic/markdown.go:129 +0x257
github.com/googleapis/gapic-generator-go/internal/gengapic.(*mdRenderer).plain(0xc0004cc000, {0x1777820?, 0xc0006a16b0?})
	/Users/chrisdsmith/oss/gapic-generator-go/internal/gengapic/markdown.go:91 +0x2d3
github.com/googleapis/gapic-generator-go/internal/gengapic.(*mdRenderer).plain(0xc0004cc000, {0x17779d0?, 0xc0004cfd40?})
	/Users/chrisdsmith/oss/gapic-generator-go/internal/gengapic/markdown.go:59 +0x8fe
github.com/googleapis/gapic-generator-go/internal/gengapic.mdPlain({0x16a2123, 0x49})
	/Users/chrisdsmith/oss/gapic-generator-go/internal/gengapic/markdown.go:35 +0x11f
github.com/googleapis/gapic-generator-go/internal/gengapic.TestMDPlain(0xc000702d00)
	/Users/chrisdsmith/oss/gapic-generator-go/internal/gengapic/markdown_test.go:85 +0xa5
testing.tRunner(0xc000702d00, 0x16c3848)
	/usr/local/go/src/testing/testing.go:1439 +0x102
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1486 +0x35f
FAIL	github.com/googleapis/gapic-generator-go/internal/gengapic	0.788s
FAIL
make: *** [test-gapic] Error 1
```


#### After

```
--- FAIL: TestMDPlain (0.00s)
    markdown_test.go:87: html <ahref="/link/to/some#thing">
         with a linebreak in tag</a> <br> test resulted in a panic: runtime error: index out of range [-1]
FAIL
FAIL	github.com/googleapis/gapic-generator-go/internal/gengapic	0.405s
FAIL
make: *** [test-gapic] Error 1
```

Comment (noahdietz) [reply internal/gengapic/markdown_test.go:86]: I see, I see...would it be too much to plumb an `error` through the call stack? WDYT?

Comment (quartzmo) [reply internal/gengapic/markdown_test.go:86]: Let me see about that! 👍

Comment (noahdietz) [reply internal/gengapic/markdown_test.go:86]: You could also consider that a separate improvement and do it in another PR

Comment (quartzmo) [reply internal/gengapic/markdown_test.go:86]: "almost done"

Comment (quartzmo) [reply internal/gengapic/markdown_test.go:86]: It turns out to be A LOT of plumbing, since the stack is pretty deep and not currently returning errors. Do you think it's a good idea to add this, or is it better to just debug the panic in case of a malformed `<a>` element?

Comment (noahdietz) [reply internal/gengapic/markdown_test.go:86]: I figured it might be gnarly...let's just debug the panic. Technically, HTML is banned in our API artifact documentation, so let's not spend more time on it. Thanks for looking anyways!

**Review by noahdietz — APPROVED** [review]

File: internal/gengapic/markdown_test.go (hunk 1 above)
Comment (quartzmo) [reply internal/gengapic/markdown_test.go]: Updated to remove the `recover`, thanks!
